miden-tx/src/tests/kernel_tests/test_tx.rs

```diff
@@ -920,17 +839,184 @@ fn test_load_foreign_account_twice() {
+        .unwrap();
+
+    let (native_account, _) = AccountBuilder::new()
+        .init_seed(ChaCha20Rng::from_entropy().gen())
+        .with_component(
+            AccountMockComponent::new_with_slots(TransactionKernel::testing_assembler(), vec![])
+                .unwrap(),
+        )
+        .nonce(ONE)
+        .build_testing()
+        .unwrap();
+
+    let mut mock_chain =
+        MockChain::with_accounts(&[native_account.clone(), foreign_account.clone()]);
+    mock_chain.seal_block(None);
+    let advice_inputs = get_mock_fpi_adv_inputs(&foreign_account, &mock_chain);
+
+    let code = format!(
+        "
+        use.std::sys
+
+        use.miden::tx
+
+        begin
+            # get the storage item at index 0
+            # pad the stack for the `execute_foreign_procedure`execution
+            padw padw padw push.0.0
+            # => [pad(14)]
+
+            # push the index of desired storage item
+            push.0
+
+            # get the hash of the `get_item` account procedure
+            push.{get_item_foreign_hash}
+
+            # push the foreign account id
+            push.{foreign_account_id}
+            # => [foreign_account_id, FOREIGN_PROC_ROOT, storage_item_index, pad(14)]
+
+            exec.tx::execute_foreign_procedure
+            # => [STORAGE_VALUE]
+
+            # assert the correctness of the obtained value
+            push.1.2.3.4 assert_eqw
+            # => []
+
+            # get the storage map at index 1
+            # pad the stack for the `execute_foreign_procedure` execution
+            padw padw push.0.0
+            # => [pad(10)]
+
+            # push the key of desired storage item
+            push.{map_key}
+
+            # push the index of desired storage item
+            push.1
+
+            # get the hash of the `get_map_item_foreign` account procedure
+            push.{get_map_item_foreign_hash}
+
+            # push the foreign account id
+            push.{foreign_account_id}
+            # => [foreign_account_id, FOREIGN_PROC_ROOT, storage_item_index, MAP_ITEM_KEY, pad(10)]
+
+            exec.tx::execute_foreign_procedure
+            # => [MAP_VALUE]
+
+            # assert the correctness of the obtained value
+            push.1.2.3.4 assert_eqw
+            # => []
+
+            # truncate the stack
+            exec.sys::truncate_stack
+        end
+        ",
+        foreign_account_id = foreign_account.id(),
+        get_item_foreign_hash = foreign_account.code().procedures()[0].mast_root(),
+        get_map_item_foreign_hash = foreign_account.code().procedures()[1].mast_root(),
+        map_key = STORAGE_LEAVES_2[0].0,
+    );
+
+    let tx_script =
+        TransactionScript::compile(code, vec![], TransactionKernel::testing_assembler()).unwrap();
+
+    let tx_context = mock_chain
+        .build_tx_context(native_account.id(), &[], &[])
+        .advice_inputs(advice_inputs.clone())
+        // .foreign_account_codes(vec![foreign_account.code().clone()])
```

Comment:
I left this line because I wanted to come to some decision about what I described in the second point of my [comment](https://github.com/0xPolygonMiden/miden-base/pull/957#issuecomment-2483135872). Because I think we can optimize the way we provide the foreign account code, so we will be able to use only `foreign_account_codes` function.